mozilla/blurts-server
# Optional field for middle name and suffix (MNTOR-2813)

# References: 
Jira: [MNTOR-2813](https://mozilla-hub.atlassian.net/browse/MNTOR-2813)
Figma: https://www.figma.com/file/DTbmXzCQCw2PxXpHQW8yfW/Monitor-MVP-Enhancements?type=design&node-id=20-1614&mode=design&t=SfweuhYdvzeoJBxW-4




# Description

This PR adds optional text fields to provide ”middle name” and “suffix” for the broker scan. The additional fields will be rolled out as an experiment on prod with [MNTOR-3085](https://mozilla-hub.atlassian.net/browse/MNTOR-3085).

# Screenshot

| Before | After |
| --- | --- |
| <img width="677" alt="mntor-2813-before" src="https://github.com/mozilla/blurts-server/assets/13835474/6e5e1492-de18-4908-844e-cf33c0c6284c"> | <img width="682" alt="mntor-2813-after" src="https://github.com/mozilla/blurts-server/assets/13835474/710e1a2e-73cd-4725-897e-b7d5a69e3533"> |


# How to test

1. Update experiments: `npm run build-nimbus`
2. Go through the “welcome scan” flow
3. Make sure running a scan works with and without the newly added optional fields and that the provided info is recorded in the `onerep_profiles` table.

# Checklist (Definition of Done)


[MNTOR-2813]: https://mozilla-hub.atlassian.net/browse/MNTOR-2813?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MNTOR-3085]: https://mozilla-hub.atlassian.net/browse/MNTOR-3085?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
